### PR TITLE
Fix an automatic aggregation bug, limit array element types that are aggregatable

### PR DESCRIPTION
**Pull request by e-kayrakli**
This PR addresses two issues with automatic aggregation.

1. I wrote a funny-looking test for another PR, and that hit a bug in automatic
   aggregation implementation.

   The bug occurs when we add an aggregator to a `forall`, then create fast/slow
   follower bodies for the said `forall`, then proceed with aggregation in one
   of those but revert aggregation in another. In this scenario, both bodies
   will be using the same aggregator symbol, and reversion of one should not
   result in removing the aggregation from the other branch.

   This bug was triggering assertions in aggregator removal code. This PR
   addresses that issue by keeping track of all the aggregator symbols that we
   end up using, and then avoiding removing them.

2. We used to only disallow arrays of classes from being aggregated. However,
   this should have been a wider net, because if a record has a `proc =`
   overload, then, aggregation will break its semantics. Note that this includes
   `string` and `bytes`, as well.

   This PR currently limits aggregation to arrays of POD types.

#### Test status
- [x] standard
- [x] gasnet
- [x] gasnet --auto-aggregation

### CODE REVIEW

**Comment by e-kayrakli** [comment]
@mppf -- this is a bug fix, and doesn't have to go in on Monday.

File: modules/internal/ChapelAutoAggregation.chpl [hunk 1]
```diff
@@ -60,9 +60,50 @@ module ChapelAutoAggregation {
...
+
+    if (dim+1 == t.size) {
+      return elemTypeSupportsAggregation(t[dim]);
+    }
+    else {
+      return elemTypeSupportsAggregation(t[dim]) &&
+             elemTypeSupportsAggregation(t, dim+1);
+
+    }
+  }
```
Comment (mppf) [inline modules/internal/ChapelAutoAggregation.chpl:102]: I don't think you should do any of that. Just use `isPOD`.

``` chapel
private proc elemTypeSupportsAggregation(type t) param {
  return isPODType(t);
}
```

See also 

https://github.com/chapel-lang/chapel/blob/e7a41c2f9891ff292173425786d9082fe23b9f47/modules/standard/Types.chpl#L129-L151

**Review by mppf — APPROVED** [review]

File: modules/internal/ChapelAutoAggregation.chpl (hunk 1 above)
Comment (ronawho) [reply modules/internal/ChapelAutoAggregation.chpl:102]: Yeah, should just use `isPOD`. I thought the unordered compiler optimization already included the pod check though.

File: test/optimizations/autoAggregation/arrTypes-badType.good [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Start analyzing forall for automatic aggregation (arrTypes.chpl:7)
+| Found an aggregation candidate (arrTypes.chpl:8)
+|  Potential source aggregation (arrTypes.chpl:8)
```
[the remark this answers is not in the document]
Comment (e-kayrakli) [reply test/optimizations/autoAggregation/arrTypes-badType.good:3]: That's right. That PR was already improving the output here, but I added another commit to improve further. With that PR in, this file will change to

```
Start analyzing forall for automatic aggregation (test/optimizations/autoAggregation/arrTypes.chpl:7)
| Found an aggregation candidate (test/optimizations/autoAggregation/arrTypes.chpl:8)
|  Potential source aggregation (test/optimizations/autoAggregation/arrTypes.chpl:8)
End analyzing forall for automatic aggregation (test/optimizations/autoAggregation/arrTypes.chpl:7)

Iterable type not suitable for aggregation (test/optimizations/autoAggregation/arrTypes.chpl:8)
Aggregation candidate has reverted local child  (test/optimizations/autoAggregation/arrTypes.chpl:8)
Could not prove the locality of either side. Will not use aggregation (test/optimizations/autoAggregation/arrTypes.chpl:8)
```

I will rebase once that PR is final and merged.

**Pull request merged**